// @dndbeyond/ttui/mui/theme/config/palette-light.ts
import { ThemeOptions } from "@mui/material";

export const palette: ThemeOptions["palette"] = {
  mode: "light",
  text: {
    primary: "#12181cff",
    secondary: "#12181ca3",
    disabled: "#12181c5c",
  },
  primary: {
    main: "#374045ff",
    dark: "#12181cff",
    light: "#75838bff",
    contrastText: "#ffffffff",
  },
  secondary: {
    main: "#e40712ff",
    dark: "#9f271dff",
    light: "#f77558ff",
    contrastText: "#ffffffff",
  },
  action: {
    active: "#12181ca3",
    hover: "#12181c14",
    selected: "#12181c1f",
    disabled: "#12181c5c",
    disabledBackground: "#12181c29",
    focus: "#12181c29",
  },
  error: {
    main: "#ed6c02ff",
    dark: "#c77700ff",
    light: "#ffb547ff",
    contrastText: "#ffffffff",
  },
  warning: {
    main: "#ed6c02ff",
    dark: "#c77700ff",
    light: "#ffb547ff",
    contrastText: "#ffffffff",
  },
  info: {
    main: "#2196f3ff",
    dark: "#0b79d0ff",
    light: "#64b6f7ff",
    contrastText: "#ffffffff",
  },
  success: {
    main: "#4caf50ff",
    dark: "#3b873eff",
    light: "#7bc67eff",
    contrastText: "#ffffffff",
  },
  background: {
    paper: "#ffffffff",
    default: "#f4f5f5ff",
    parchment: "#f9f6efff",
    scroll: "#eee8dbff",
  },
  common: { white: "#ffffffff", black: "#000000ff" },
  grey: {
    "50": "#f4f5f5ff",
    "100": "#ecedeeff",
    "200": "#dcdfe1ff",
    "300": "#c4cbceff",
    "400": "#a2acb2ff",
    "500": "#75838bff",
    "600": "#525c63ff",
    "700": "#374045ff",
    "800": "#232b2fff",
    "900": "#12181cff",
  },
  rarity: {
    uncommon: "#009158ff",
    rare: "#077bd0ff",
    veryRare: "#9605d4ff",
    legendary: "#ae5e21ff",
    artifact: "#c81e18ff",
    contrastText: "#ffffffff",
  },
  reference: {
    character: "#ee8600ff",
    magicItem: "#077bd0ff",
    monster: "#c60000ff",
    skill: "#009158ff",
    spell: "#9605d4ff",
    contrastText: "#ffffffff",
  },
};
